src/client/ua_client_highlevel_subscriptions.c

```diff
@@ -236,8 +313,9 @@ UA_Client_processPublishResponse(UA_Client *client, UA_PublishRequest *request,
     if(!sub)
         return;
 
-    UA_LOG_DEBUG(client->config.logger, UA_LOGCATEGORY_CLIENT,
-                 "Processing a publish response on subscription %u with %u notifications",
+    if ((unsigned int)response->notificationMessage.notificationDataSize > 0)
```

Comment:
Use UA datatypes instead of `unsigned int`. E.g. `UA_UInt32`.
Use logging functions instead of printf